src/wp-settings.php

```diff
@@ -503,68 +503,71 @@
 $failed_plugins     = array();
 $plugins_dir_strlen = strlen( trailingslashit( WP_PLUGIN_DIR ) );
 foreach ( wp_get_active_and_valid_plugins() as $plugin ) {
-	$plugin_file    = substr( $plugin, $plugins_dir_strlen );
-	$plugin_headers = $all_plugin_data[ $plugin_file ];
-	$errors         = array();
-	$requirements   = array(
-		'requires'     => ! empty( $plugin_headers['RequiresWP'] ) ? $plugin_headers['RequiresWP'] : '',
-		'requires_php' => ! empty( $plugin_headers['RequiresPHP'] ) ? $plugin_headers['RequiresPHP'] : '',
-	);
-	$compatible_wp  = is_wp_version_compatible( $requirements['requires'] );
-	$compatible_php = is_php_version_compatible( $requirements['requires_php'] );
-
-	$php_update_message = '</p><p>' . sprintf(
-		/* translators: %s: URL to Update PHP page. */
-		__( '<a href="%s">Learn more about updating PHP</a>.' ),
-		esc_url( wp_get_update_php_url() )
-	);
+	$plugin_file = substr( $plugin, $plugins_dir_strlen );
+
+	if ( isset( $all_plugin_data[ $plugin_file ] ) ) {
```

Comment:
If (for any reason) the plugin is not in the array or parts of it data does not exist within it, this check is essential.

I wonder though - for those unfamiliar with this code, looking at this, it might not be clear why this guard is needed. 

@costdev what do you think about adding an inline comment?

Reply:
More context: I would have expected the data to be there for each of the plugins to be loaded. So this bug is a TIL for me.

Reply:
There's no data if the plugin is added via FTP for example.

Reply:
Does such a comment work?

```
// Safeguard if the plugin data is not already stored, e.g. because it was uploaded manually.
```

Reply:
Maybe:

```
// Safeguard if the plugin data is not already stored, e.g. it was uploaded manually or via wp-cli scaffolding.
```